runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/SourceInputFormat.java

```diff
@@ -73,8 +69,9 @@ public void configure(Configuration configuration) {}
 
   @Override
   public void open(SourceInputSplit<T> sourceInputSplit) throws IOException {
+    options = new ObjectMapper().readValue(serializedOptions, PipelineOptions.class);
```

Comment:
Good point. This will be called on every input split. We can move the deserialization code to the configure method.